core/graph/src/main/java/uk/gov/gchq/gaffer/graph/Graph.java

```diff
@@ -995,6 +1000,11 @@ private void updateGraphHooks(final GraphConfig config) {
                     config.getHooks().add(0, new NamedOperationResolver());
                 }
             }
+            if (!hasFunctionAuthoriserHook && !config.isSkipDefaultSecurityHooks()) {
```

Comment:
If we are adding this FunctionAuthoriser by default for every Gaffer system, we should make sure it is efficient and doesn't error.